src/txkube/test/test_authentication.py

```diff
@@ -68,6 +72,35 @@
 -----END CERTIFICATE-----
 """
 
+
+def createReactor():
+    """
+    Twisted 17.1.0 and higher requires a reactor which implements
+    ``IReactorPluggableNameResolver``.
+    """
+
+    class Resolution(object):
+        name = None
+
+    class _FakeResolver(object):
+        def resolveHostName(self, resolutionReceiver, hostName, *args,  **kwargs):
+            portNumber = kwargs.pop('portNumber')
+            r = Resolution()
+            r.name = hostName
```

Comment:
How about `Resolution(name=hostName)`?  Your choice of `pyrsistent` or `attrs` for defining `Resolution` to make this possible. :)